### DISCUSSION TITLE
Improve translations workflow in SDK

### DISCUSSION THREAD

**Discussion by pavelstianko** [directus/directus#20523] [Feature Requests]
### Summary

If you are working with translations, it would be better to use a parameter for the desired language. Currently, you have to use deep and filter the results. When working on a multilingual website, this is a lot of work, and sometimes it can be hard to read the code.

### Basic Example

It would be nice to have something like this:

readItems('articles', {
		fields: ['id', 'title', translation{'translated-field'}],
		translation: 'en-US'
})

or 

readItems('articles', {
		fields: ['id', 'title', translation('en-US'){'translated-field'}]
})

### Motivation

It will speed up the language handling process, especially when modifying a website to support multiple languages.

### Detailed Design

Add translation parameter to SDK, so you can use translated fields almost like normal fields.

### Must Have:
Parameter for language in SDK

### Should Have:
Easy syntax so you can work with translated fields almost like with normal fields

### Drawbacks

It could be hard to implement it this way.

### Alternatives

There can another syntax for doing the same.

### Adoption Strategy

It is new feature without any breakinng change.

**Comment by br41nslug** [comment]
This should probably be implemented in the API, rather than purely on the SDK side (which is not possible). Looks like a good improvement on DX at a (hopefully small) performance cost of having to walk the schema for translations.

**Reply by pavelstianko** [reply]
Yes, that would be nice. Stripe is using GET /api/{content-type}?locale={locale-code}

**Comment by vitalijalbu** [comment]
I'm also looking fot this feature, like Strapi...

**Comment by MarvinXu** [comment]
Glad to see someone raise this! It has bothered me for a long time. What is your current workflow? @pavelstianko 

What I do now is getting all translations when fetching data, then define a `getXXXByLang` function to flatten the data. Looking for a more elegant way~
```js
function getTranslation(translations, lang) {
  const data = translations.find(t => t.languages_code === locales[lang]);
  return data && Object.keys(data).reduce((prev, key) => {
    if (!key.endsWith("_id") && !["id", "languages_code"].includes(key)) {
      prev[key] = data[key];
    }
    return prev;
  }, {});
}

function getProductByLang(product, lang) {
  return {
    ...getTranslation(product.translations, lang),
    features: product.features.map(feature => ({
      ...getTranslation(feature.translations, lang)
    }))
  }
}
```

**Reply by pavelstianko** [reply]
Hello Marvin, now I am using filter in request, but if you have deeps with own translations as well it is starting to be a little bit confusing. Now I moved from SDK to to GraphQL where is deep and filter easier and better to read with complicated fetch.

**Reply by MarvinXu** [reply]
Thanks! I'll check on GraphQL.

**Comment by gastonbesada** [comment]
Maybe duplicated: https://github.com/directus/directus/discussions/6654

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
